react/features/chat/middleware.js

```diff
@@ -116,44 +152,183 @@ function _addChatMsgListener(conference, { dispatch, getState }) {
     conference.on(
         JitsiConferenceEvents.MESSAGE_RECEIVED,
         (id, message, timestamp, nick) => {
-            // Logic for all platforms:
-            const state = getState();
-            const { isOpen: isChatOpen } = state['features/chat'];
-
-            if (!isChatOpen) {
-                dispatch(playSound(INCOMING_MSG_SOUND_ID));
-            }
+            _handleReceivedMessage({
+                dispatch,
+                getState
+            }, {
+                id,
+                message,
+                nick,
+                privateMessage: false,
+                timestamp
+            });
+        }
+    );
 
-            // Provide a default for for the case when a message is being
-            // backfilled for a participant that has left the conference.
-            const participant = getParticipantById(state, id) || {};
-            const displayName = participant.name || nick || getParticipantDisplayName(state, id);
-            const hasRead = participant.local || isChatOpen;
-            const timestampToDate = timestamp
-                ? new Date(timestamp) : new Date();
-            const millisecondsTimestamp = timestampToDate.getTime();
-
-            dispatch(addMessage({
-                displayName,
-                hasRead,
+    conference.on(
+        JitsiConferenceEvents.PRIVATE_MESSAGE_RECEIVED,
```

Comment:
There will be a time potentially where older mobile clients wont be seeing private messages sent by clients that have private message abilities. I'm not sure anything can or should be done about it though.